graphql-ballerina/engine.bal

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import graphql.parser;
+
+# Represents the Ballerina GraphQL engine.
+public class Engine {
+    private Listener 'listener;
+    private __Schema? schema;
```

Comment:
Any especial reason to use __Schema instead of Schema ?

Reply:
Yes, the GraphQL introspections types use `__` for their internal types. This is added to make it easy for implementing the introspection support.